src/games/strategy/engine/framework/GameRunner.java

```diff
@@ -785,4 +574,217 @@ public static Image getGameIcon(final Window frame) {
+  }
+
+  public static void startGame(final String savegamePath, final String classpath, final Long maxMemory) {
+    final List<String> commands = new ArrayList<>();
+    if (maxMemory != null && maxMemory > (32 * 1024 * 1024)) {
+      ProcessRunnerUtil.populateBasicJavaArgs(commands, classpath, maxMemory);
+    } else {
+      ProcessRunnerUtil.populateBasicJavaArgs(commands, classpath);
+    }
+    if (savegamePath != null && savegamePath.length() > 0) {
+      commands.add("-D" + GameRunner.TRIPLEA_GAME_PROPERTY + "=" + savegamePath);
+    }
+    // add in any existing command line items
+    for (final String property : GameRunner.COMMAND_LINE_ARGS) {
+      // we add game property above, and we add version bin in the populateBasicJavaArgs
+      if (GameRunner.TRIPLEA_GAME_PROPERTY.equals(property)
+          || GameRunner.TRIPLEA_ENGINE_VERSION_BIN.equals(property)) {
+        continue;
+      }
+      final String value = System.getProperty(property);
+      if (value != null) {
+        commands.add("-D" + property + "=" + value);
+      } else if (GameRunner.LOBBY_HOST.equals(property) || LobbyServer.TRIPLEA_LOBBY_PORT_PROPERTY.equals(property)
+          || GameRunner.LOBBY_GAME_HOSTED_BY.equals(property)) {
+        // for these 3 properties, we clear them after hosting, but back them up.
+        final String oldValue = System.getProperty(property + GameRunner.OLD_EXTENSION);
+        if (oldValue != null) {
+          commands.add("-D" + property + "=" + oldValue);
+        }
+      }
+    }
+    // classpath for main
+    commands.add(GameRunner.class.getName());
+    ProcessRunnerUtil.exec(commands);
+  }
+
+  public static void hostGame(final int port, final String playerName, final String comments, final String password,
+      final Messengers messengers) {
+    final List<String> commands = new ArrayList<>();
+    ProcessRunnerUtil.populateBasicJavaArgs(commands);
+    commands.add("-D" + TRIPLEA_SERVER_PROPERTY + "=true");
+    commands.add("-D" + TRIPLEA_PORT_PROPERTY + "=" + port);
+    commands.add("-D" + TRIPLEA_NAME_PROPERTY + "=" + playerName);
+    commands.add("-D" + LOBBY_HOST + "="
+        + messengers.getMessenger().getRemoteServerSocketAddress().getAddress().getHostAddress());
+    commands
+        .add("-D" + LobbyServer.TRIPLEA_LOBBY_PORT_PROPERTY + "=" + messengers.getMessenger().getRemoteServerSocketAddress().getPort());
+    commands.add("-D" + LOBBY_GAME_COMMENTS + "=" + comments);
+    commands.add("-D" + LOBBY_GAME_HOSTED_BY + "=" + messengers.getMessenger().getLocalNode().getName());
+    if (password != null && password.length() > 0) {
+      commands.add("-D" + TRIPLEA_SERVER_PASSWORD_PROPERTY + "=" + password);
+    }
+    final String fileName = System.getProperty(TRIPLEA_GAME_PROPERTY, "");
+    if (fileName.length() > 0) {
+      commands.add("-D" + TRIPLEA_GAME_PROPERTY + "=" + fileName);
+    }
+    final String javaClass = "games.strategy.engine.framework.GameRunner";
```

Comment:
I believe you mean it should *not* be an unchecked String. I very much agree that is quite bad. Addressed in: 37e1d96c78dbfe01b595ff199bce3b0762205f64